sidestep_engine/vendor/data_module.py

```
@@ -308,8 +408,21 @@ def __getitem__(self, idx: int) -> Dict[str, torch.Tensor]:
         target_latents = data["target_latents"]      # [T, 64]
         attention_mask = data["attention_mask"]        # [T]
         context_latents = data["context_latents"]      # [T, 65]
-        encoder_hidden_states = data["encoder_hidden_states"]  # [L, D]
-        encoder_attention_mask = data["encoder_attention_mask"]  # [L]
+
+        # Dynamic genre/caption selection: pick genre variant with
+        # probability genre_ratio/100 when available in the .pt file.
+        use_genre = (
+            self._genre_ratio > 0
+            and "encoder_hidden_states_genre" in data
+            and random.randint(1, 100) <= self._genre_ratio
```

Ruff (0.15.7)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)